javaslang/src/main/java/javaslang/collection/Traversable.java

```diff
@@ -1203,4 +1219,19 @@ default Number sum() {
      * @return A new traversable containing all elements of this traversable paired with their index, starting with 0.
      */
     Traversable<Tuple2<T, Long>> zipWithIndex();
+
+    /**
+     * Returns a traversable formed from this traversable and another Iterable collection by mapping elements.
+     * If one of the two iterables is longer than the other, its remaining elements are ignored.
+     * <p>
+     * The length of the returned traversable is the minimum of the lengths of this traversable and {@code that}
+     * iterable.
+     *
+     * @param <U>    The type of the mapped elements.
+     * @param mapper a mapper.
```

Comment:
We could fix the descriptions of map, flatMap, zipWithIndex et.al. in a separate PR